engine/src/main/java/org/terasology/rendering/nui/layouts/FlowLayout.java

```diff
@@ -83,7 +92,7 @@ public Vector2i getPreferredContentSize(Canvas canvas, Vector2i sizeHint) {
             }
         }
         result.x = Math.max(result.x, filledWidth);
-        result.y += filledHeight;
+        result.y += filledHeight + verticalSpacing * (contents.size() - 1);
```

Comment:
Good point - that's something I missed before 👍 

I think you can only multiply with the number of elements in the content if you can make sure that there is only one element per row. However, if you make the flow are bigger, you should see that there might be multiple widgets in the same row before the automatic wrapping kicks in.

Reply:
Indeed, this is a similar note with the one for adding the spacing in the first place, should have thought about that. I'll look into changing `contents.size() - 1` with a variable that counts the rows of the FlowLayout!

Reply:
I think it might be enough to just increment `result.y` by `verticalSpacing` whenever the spacing is applied to the offset.